File: Ruby_2Darbas/CIKLAI/trecias.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    float a,a1,e = 0.00001,n,n1;
    n=1;
    n1=n+1;
    do
    {
        a=(n+1)/(sqrt((n+1)*(n+1)+1)*(sqrt((n+1)*(n+1)-1)));
        a1=(n1+1)/(sqrt((n1+1)*(n1+1)+1)*(sqrt((n1+1)*(n1+1)-1)));
        n++;
        n1++;
    }   while (abs(a1-a)>=e);
    cout<<"a["<<n<<"]: "<<a<<endl;
    cout<<"a["<<n1<<"]: "<<a1<< endl;
    cout<<"riba yra tarp "<<n<<" ir "<<n1<<" nariu."<<endl;
    //c = abs(a - a1);
    //cout << c << endl << endl;
    //cout<<fixed<<setprecision(5)<<e<<endl<<endl;
    /*int n;
    cout<<"iveskite n"<<endl;
    cin>>n;
    float c[n];
    c[1] = (n+1)/((sqrt(pow((n+1),2)+1))(sqrt(pow((n+1),2)-1)));
    float b = c[n+1];
    cout << "a[n+1] yra: " << b << endl;
    */
return 0;
}

File: Ruby_2Darbas/CIKLAI/Ruby/trecias.rb
# frozen_string_literal: true

e = 0.00001
n = 1
n1 = n + 1
a = 0
a1 = 0
loop do
  b = Math.sqrt((n + 1) * (n + 1) + 1) * Math.sqrt((n + 1) * (n + 1) - 1)

  a = (n + 1) / b
  b1 = Math.sqrt((n1 + 1) * (n1 + 1) + 1) * Math.sqrt((n1 + 1) * (n1 + 1) - 1)
  a1 = (n1 + 1) / b1
  n += 1
  n1 += 1
  break if (a1 - a).abs < e
end

puts "a[#{n}]: #{a}"
puts "a[#{n1}]: #{a1}"
puts "riba yra tarp #{n} ir #{n1} nariu."
